implementation/src/test/java/io/smallrye/jwt/auth/principal/KeyLocationResolverTest.java

```diff
@@ -0,0 +1,72 @@
+/*
+ *   Copyright 2019 Red Hat, Inc, and individual contributors.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+package io.smallrye.jwt.auth.principal;
+
+import static java.util.Collections.emptyList;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.security.PublicKey;
+
+import org.jose4j.jwk.JsonWebKey;
+import org.jose4j.jws.JsonWebSignature;
+import org.jose4j.jwx.Headers;
+import org.jose4j.lang.UnresolvableKeyException;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.internal.util.reflection.FieldReader;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class KeyLocationResolverTest {
+
+    @Mock
+    JsonWebSignature jsonWebSignature;
+
+    @Rule
+    public ExpectedException expectedEx = ExpectedException.none();
+
+    @Test
+    public void testLoadingPublicKeyWithWrongResourceLocation() throws Exception {
+
+        expectedEx.expect(UnresolvableKeyException.class);
+
+        KeyLocationResolver keyLocationResolver = new KeyLocationResolver(new JWTAuthContextInfo("wrong_location.pem", null));
+        keyLocationResolver.resolveKey(jsonWebSignature, emptyList());
+    }
+
+    @Test
+    public void testHttpsKeyLocation() throws Exception {
+        JWTAuthContextInfo contextInfo = new JWTAuthContextInfo("https://github.com/my_key.pem", "issuer");
+        contextInfo.setJwksRefreshInterval(10);
+        Headers headers = Mockito.mock(Headers.class);
+        Mockito.when(headers.getStringHeaderValue(JsonWebKey.KEY_ID_PARAMETER)).thenReturn("1");
+        JsonWebSignature signature = Mockito.mock(JsonWebSignature.class);
+        Mockito.when(signature.getHeaders()).thenReturn(headers);
+        PublicKey key = Mockito.mock(PublicKey.class);
+        KeyLocationResolver keyLocationResolver = new KeyLocationResolver(contextInfo);
+        keyLocationResolver = Mockito.spy(keyLocationResolver);
+        Mockito.doReturn(key).when(keyLocationResolver).getHttpJwk(Mockito.any(), Mockito.any());
+        assertEquals(key, keyLocationResolver.resolveKey(signature, emptyList()));
+        assertTrue(
+                new FieldReader(keyLocationResolver, KeyLocationResolver.class.getDeclaredField("verificationKey")).isNull());
```

Comment:
Looking at the package name, `FieldReader` doesn't seem to be considered as public API by Mockito.

Reply:
I suggest making `verificationKey` package private so that it can be accessed by unit tests without reflection.

Reply:
i made `verificationKey` package private